pkg/mirrorregistries/generator.go

```diff
@@ -69,6 +70,18 @@ func (m *mirrorRegistriesConfigBuilder) GetMirrorRegistries() ([]byte, error) {
 	return os.ReadFile(m.MirrorRegistriesConfigPath)
 }
 
+// Determines whether or not this installation of assisted is using mirror registries
+func (m *mirrorRegistriesConfigBuilder) HasMirrorRegistries() (bool, error) {
+	hasMirrorRegistries := false
+	if _, err := m.GetMirrorRegistries(); err != nil {
+		if !os.IsNotExist(err) {
+			return false, err
```

Comment:
maybe just return hasMirrorRegistries ? as it should be false in anycase? just for consistency

Reply:
I mean, that assumes you just want to swallow that error, I don't think that's our decision to make.

What if there is some error, unrelated to the configuration of mirror registries, maybe some disk IO problem or something? We would just say "there are no mirror registries" instead of saying "I encountered an unexpected error"?

I think we are right to handle in the way I have chosen, any errors we do not expect, should be propagated upward and handled.

Reply:
OK, I went with your suggestion, though it does mean that we are swallowing errors in some cases.

A problem fetching mirror registries does not necessarily imply that we should treat this scenario as hasMirrorRegistries := false

Saves a few lines of code, looks more elegant but is that really just an illusion?

Reply:
Just found `r.MirrorRegistriesConfigBuilder.IsMirrorRegistriesConfigured()` which looks a lot more appropriate!, going to scrap this function altogether.